packages/block-editor/src/store/actions.js

```diff
@@ -374,6 +374,23 @@ export function mergeBlocks( firstBlockClientId, secondBlockClientId ) {
 	};
 }
 
+/**
+ * Returns action objects used in signalling that the blocks corresponding to
+ * the set of specified client IDs are to be removed.
+ * This action does not trigger any required side effects and it is not recommended for public usage.
+ *
+ * @param {string|string[]} clientIds      Client IDs of blocks to remove.
+ *
+ * @return {Object} Action object.
+ *
+ */
+export function __internalRemoveBlocksPure( clientIds ) {
```

Comment:
I don't understand why this needs to exist. What was the issue with the previous `yield { type: 'REMOVE_BLOCKS' }` ?

Reply:
If we do that the yield select that appears after will return the value before the store is changed.

Reply:
This solution seems a little bit hacky, I'm probably missing something but it seemed the only way I could select from the store after the action changed the store. If there is a simpler method that I'm missing I will gladly change the code.

Reply:
> If we do that the yield select that appears after will return the value before the store is changed.

Then if we're at the moment prior to removal, would it be enough to test that the count is `=== 1` (or I guess `=== clientIds.length`), assuming that it'd become zero after the removal?

An issue with this may be that there's nothing to stop me from dispatching this action with a clientId which doesn't actually exist, at which point there's still one block after calling removeBlocks

Reply:
> Then if we're at the moment prior to removal, would it be enough to test that the count is `=== 1` (or I guess `=== clientIds.length`), assuming that it'd become zero after the removal?

This logic would be complex.
If I have a columns block with lots of blocks inside `wp.data.select( 'core/block-editor' ).getBlockCount() === 1 `because only columns block exists at the root level, I may remove one more blocks inside the columns and getBlockCount() will continue to return 1.
We would need checks to understand if the block being removed is the one at the root level, in this case, this checks would not be very hard. But, my plan is to use this approach in other PR I have for the block replace case and there checks are more complex.
Ideally, we would have a solution that allows side effects depending on the store before the store is changed and after the store is changed without checks, similar to what a middleware allows.

Reply:
>Ideally, we would have a solution that allows side effects depending on the store before the store is changed and after the store is changed without checks, similar to what a middleware allows.

Yeah, in fact, re-reading your original remark, this seems like buggy behavior to me. The result of `select` seems like it should be accurate at the point in time at which the control was yielded.

>If we do that the yield select that appears after will return the value before the store is changed.